librpc/src/rpc_thread_factory.cc

```diff
@@ -104,18 +104,23 @@ class rpc_thread : public Thread {
     }
 
     pthread_attr_t thread_attr;
-    if (pthread_attr_init(&thread_attr) != 0) {
-      throw SystemResourceException("pthread_attr_init failed");
+    auto ret_init = pthread_attr_init(&thread_attr);
+    if (ret_init != 0) {
+      std::string errMsg = "pthread_attr_init failed: " + std::to_string(ret_init);
```

Comment:
Confluo does not use camel case for variables. In fact, in this case, it might be better not to use the variable at all, i.e., pass the message directly to the exception constructor.